Bump version to v0.24.0 (#1533)
open-mmlab/mmsegmentation: docs/en/changelog.md

- Avoid converting label ids twice by label map during evaluation ([1417](https://github.com/open-mmlab/mmsegmentation/pull/1417))
- Fix bug about label_map ([1445](https://github.com/open-mmlab/mmsegmentation/pull/1445))
- Fix image save path bug in Windows ([1423](https://github.com/open-mmlab/mmsegmentation/pull/1423))
- Fix MMSegmentation Colab demo ([1501](https://github.com/open-mmlab/mmsegmentation/pull/1501), [1452](https://github.com/open-mmlab/mmsegmentation/pull/1452))
- Migrate azure blob for beit checkpoints ([1503](https://github.com/open-mmlab/mmsegmentation/pull/1503))
- Fix bug in `tools/analyse_logs.py` caused by wrong plot_iter in some cases ([1428](https://github.com/open-mmlab/mmsegmentation/pull/1428))

**Improvements**

- Merge BEiT and ConvNext's LR decay optimizer constructors ([#1438](https://github.com/open-mmlab/mmsegmentation/pull/1438))
- Register optimizer constructor with mmseg ([#1456](https://github.com/open-mmlab/mmsegmentation/pull/1456))
- Refactor transfomer encode layer in ViT and BEiT backbone ([#1481](https://github.com/open-mmlab/mmsegmentation/pull/1481))

> - Add build_pos_embed and build_layers for BEiT ([1517](https://github.com/open-mmlab/mmsegmentation/pull/1517))

Reviewer: ```suggestion
- Add `build_pos_embed` and `build_layers` for BEiT ([1517](https://github.com/open-mmlab/mmsegmentation/pull/1517))
```

> - Add with cp to mit and vit ([1431](https://github.com/open-mmlab/mmsegmentation/pull/1431))

Reviewer: ```suggestion
- Add `with_cp` to mit and vit ([1431](https://github.com/open-mmlab/mmsegmentation/pull/1431))
```

> - Fix inconsistent dtype of seg_label in stdc decode ([1463](https://github.com/open-mmlab/mmsegmentation/pull/1463))

Reviewer: ```suggestion
- Fix inconsistent dtype of `seg_label` in stdc decode ([1463](https://github.com/open-mmlab/mmsegmentation/pull/1463))
```

> - Delete seed in dist_train.sh ([1519](https://github.com/open-mmlab/mmsegmentation/pull/1519))

Reviewer: ```suggestion
- Delete random seed for training in `dist_train.sh` ([1519](https://github.com/open-mmlab/mmsegmentation/pull/1519))
```

- Revise high `workers_per_gpus` in config file ([#1506](https://github.com/open-mmlab/mmsegmentation/pull/1506))
- Add GPG keys and del mmcv version in Dockerfile ([1534](https://github.com/open-mmlab/mmsegmentation/pull/1534))
- Update checkpoint for model in deeplabv3plus ([#1487](https://github.com/open-mmlab/mmsegmentation/pull/1487))

> - Add DistSamplerSeedHook for when runner is EpochBasedRunner ([1449](https://github.com/open-mmlab/mmsegmentation/pull/1449))

Reviewer: ```suggestion
- Add `DistSamplerSeedHook` to set epoch num to dataloader when runner is `EpochBasedRunner` ([1449](https://github.com/open-mmlab/mmsegmentation/pull/1449))
```
